code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -200,28 +200,30 @@
 		target_types += /obj/effect/decal/cleanable/crayon
 
 	if(trash)
-		target_types += /obj/item/trash
+		target_types = list(
+		/obj/item/trash,
+		/obj/item/reagent_containers/food/snacks/deadmouse
+		)
 
 	target_types = typecacheof(target_types)
 
 /mob/living/simple_animal/bot/cleanbot/UnarmedAttack(atom/A)
-	if(istype(A, /obj/effect/decal/cleanable))
-		anchored = TRUE
+	if(is_cleanable(A))
 		icon_state = "cleanbot-c"
-		visible_message("<span class='notice'>[src] begins to clean up [A].</span>")
 		mode = BOT_CLEANING
-		spawn(50)
-			if(mode == BOT_CLEANING)
-				if(A && isturf(A.loc))
-					var/atom/movable/AM = A
-					if(istype(AM, /obj/effect/decal/cleanable))
-						for(var/obj/effect/decal/cleanable/C in A.loc)
-							qdel(C)
-
-				anchored = FALSE
-				target = null
-			mode = BOT_IDLE
-			icon_state = "cleanbot[on]"
+
+		var/turf/T = get_turf(A)
+		if(do_after(src, 1, target = T))
+			SEND_SIGNAL(T, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_MEDIUM)
+			for(var/atom/dirtything in T)
+				if(is_cleanable(dirtything))
+					qdel(dirtything)
+
+		visible_message("<span class='notice'>[src] cleans up [A].</span>")
```

Comment:
This needs to be moved up by a few lines. Right under the signal (line 217) would be perfect. Otherwise this will get displayed even if the act of cleaning fails.

I might also needs a QDELETED check in case it gets deleted in that one decisecond.

Something like
```
QDELETED(A) ? visible_message("<span class='notice'>[src] cleans up [A].</span>") :  visible_message("<span class='notice'>[src] cleans \the [T].</span>")